packages/block-library/src/list-item/edit.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	RichText,
+	useBlockProps,
+	useInnerBlocksProps,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+export default function ListItemEdit( { attributes, setAttributes } ) {
+	const blockProps = useBlockProps();
+	const innerBlocksProps = useInnerBlocksProps( blockProps, {
+		allowedBlocks: [ 'core/list' ],
+	} );
```

Comment:
Nitpick: `{ allowedBlocks: [ 'core/list' ] }` could be moved to a constant outside of the component to avoid recreating it on every render.